js/page.js

```diff
@@ -0,0 +1,30 @@
+(function () {
+    const iframe = document.querySelector('.s-page-summary iframe');
+    if (!iframe) return; // no iframe detected
+
+    const pageTitle = document.querySelector('.s-page-title');
+    const link = createElement("a", [], {
+        href: iframe.src,
+        title: 'Open in a new tab',
+        target: '_blank',
+        style: 'background: none;' // clear assignment icon
+    }, [pageTitle.cloneNode(true), createElement('span', [], {
+        textContent: ' ⇨'
+    })]);
+    // turn span to link element
+    pageTitle.replaceWith(link);
+
+    link.href = processIframeSrc(link);
+})();
+
+function processIframeSrc(link) {
+    // Google docs/sheets/slides
+    if (link.host.match(/.*s\.google\.com/g)) {
+        // yeah yeah you could do an extensive regex match OR you can just be lazy
```

Comment:
I'm still not super keen on the matching/replacement logic here. Matching specifically on `pub?embedded` as a string literal seems very fragile (what if there's another query parameter?) - I'd be okay with stripping the `embedded` parameter from any GDocs URL.

I'm also okay with eschewing hard-to-read regexes - as I mentioned in another comment, there should be a URL parsing API you can use. Once you use that, semantically meaningful query string manipulation should be fairly straightforward (or, at worst, doable with a not-bad regex, although I'd be shocked if there isn't a builtin component for doing almost exactly what we want). Likewise, if you can have the browser isolate the path component of the URL for you, using something like
```js
pathComponent.replace(/preview(\/?)$/, "/edit$1")
```
should be fairly trivial.

Reply:
Yeah, I believe [URL.pathname](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname) is what you're looking for here